### PR TITLE
Tarjan's Algorithm added

### PR DESCRIPTION
**Pull request by atharvagadkari05**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added Tarjan's algorithm and made necessary changes in the readme file.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1736

### CODE REVIEW

File: Java/README.md [hunk 1]
```diff
@@ -15,14 +15,14 @@
 - [Russian Peasant Algorithm](Algorithms/RussianPeasant.java)
 - [Z-Algorithm](Algorithms/ZAlgorithm.java)
 - [Moore Algorithm](Algorithm/Moore_Algorithm.java)
-
+- [Tarjan's Algorithm](Algorithm/TarjansAlgorithm.java)
```
Comment (atarax665) [inline Java/README.md:18]:
```suggestion
- [Tarjan's Algorithm](Algorithm/TarjansAlgorithm.java)

```

File: Java/Algorithms/TarjansAlgorithm.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
+import java.io.*;
```
Comment (atarax665) [inline Java/Algorithms/TarjansAlgorithm.java:1]: Add beginner introduction of the algorithm with space and time complexity.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
Resolve the merge conflict.

File: Java/Algorithms/TarjansAlgorithm.java (hunk 2 above)
Comment (atharvagadkari05) [reply Java/Algorithms/TarjansAlgorithm.java:1]: Can you please elaborate on it??

**Comment by HarshCasper** [comment]
Closing the PR because of no updates.

**Pull request closed without merging**